packages/dx-react-chart/src/plugins/tooltip.jsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {
+  Plugin,
+  Getter,
+  TemplateConnector,
+  Template,
+  TemplatePlaceholder,
+  withComponents,
+} from '@devexpress/dx-react-core';
+import { getParameters, processPointerMove } from '@devexpress/dx-chart-core';
+import { Target } from '../templates/tooltip/target';
+
+class RawTooltip extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      target: props.targetItem || props.defaultTargetItem,
+    };
+    const handlePointerMove = this.handlePointerMove.bind(this);
+    this.getPointerMoveHandlers = ({ pointerMoveHandlers }) => [
+      ...pointerMoveHandlers, handlePointerMove,
+    ];
+  }
+
+  handlePointerMove({ targets }) {
+    const { onTargetItemChange } = this.props;
+    const { target: currentTarget } = this.state;
+    const target = processPointerMove(targets, currentTarget, onTargetItemChange);
+    if (target !== undefined) {
+      if (target && target.point === undefined) {
+        this.setState({
```

Comment:
Since calculated state depends on current state callback version of `setState` should be used.

Reply:
done